// Code generated by mockery v2.10.0. DO NOT EDIT.

package mocks

import (
	context "context"

	model "github.com/soundrussian/go-practicum-diploma/model"
	mock "github.com/stretchr/testify/mock"
)

// Storage is an autogenerated mock type for the Storage type
type Storage struct {
	mock.Mock
}

// Close provides a mock function with given fields:
func (_m *Storage) Close() {
	_m.Called()
}

// CreateUser provides a mock function with given fields: ctx, login, password
func (_m *Storage) CreateUser(ctx context.Context, login string, password string) (*model.User, error) {
	ret := _m.Called(ctx, login, password)

	var r0 *model.User
	if rf, ok := ret.Get(0).(func(context.Context, string, string) *model.User); ok {
		r0 = rf(ctx, login, password)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*model.User)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, string, string) error); ok {
		r1 = rf(ctx, login, password)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// FetchUser provides a mock function with given fields: ctx, login
func (_m *Storage) FetchUser(ctx context.Context, login string) (*model.User, error) {
	ret := _m.Called(ctx, login)

	var r0 *model.User
	if rf, ok := ret.Get(0).(func(context.Context, string) *model.User); ok {
		r0 = rf(ctx, login)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*model.User)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, string) error); ok {
		r1 = rf(ctx, login)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// UserBalance provides a mock function with given fields: ctx, userID
func (_m *Storage) UserBalance(ctx context.Context, userID uint64) (*model.UserBalance, error) {
	ret := _m.Called(ctx, userID)

	var r0 *model.UserBalance
	if rf, ok := ret.Get(0).(func(context.Context, uint64) *model.UserBalance); ok {
		r0 = rf(ctx, userID)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*model.UserBalance)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, uint64) error); ok {
		r1 = rf(ctx, userID)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// UserWithdrawals provides a mock function with given fields: ctx, userID
func (_m *Storage) UserWithdrawals(ctx context.Context, userID uint64) ([]model.Withdrawal, error) {
	ret := _m.Called(ctx, userID)

	var r0 []model.Withdrawal
	if rf, ok := ret.Get(0).(func(context.Context, uint64) []model.Withdrawal); ok {
		r0 = rf(ctx, userID)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).([]model.Withdrawal)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, uint64) error); ok {
		r1 = rf(ctx, userID)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}

// Withdraw provides a mock function with given fields: ctx, userID, withdrawal
func (_m *Storage) Withdraw(ctx context.Context, userID uint64, withdrawal model.Withdrawal) (*model.Withdrawal, error) {
	ret := _m.Called(ctx, userID, withdrawal)

	var r0 *model.Withdrawal
	if rf, ok := ret.Get(0).(func(context.Context, uint64, model.Withdrawal) *model.Withdrawal); ok {
		r0 = rf(ctx, userID, withdrawal)
	} else {
		if ret.Get(0) != nil {
			r0 = ret.Get(0).(*model.Withdrawal)
		}
	}

	var r1 error
	if rf, ok := ret.Get(1).(func(context.Context, uint64, model.Withdrawal) error); ok {
		r1 = rf(ctx, userID, withdrawal)
	} else {
		r1 = ret.Error(1)
	}

	return r0, r1
}
